overlord/snapstate/backend/link.go

```diff
@@ -299,9 +293,15 @@ func (b Backend) QueryDisabledServices(info *snap.Info, pb progress.Meter) ([]st
 	return wrappers.QueryDisabledServices(info, pb)
 }
 
-func removeCurrentSymlinks(info snap.PlaceInfo) error {
+func removeCurrentSymlinks(info *snap.Info, firstInstall bool) error {
 	var err1, err2 error
 
+	if info.Type() == snap.TypeSnapd && !firstInstall {
```

Comment:
in theory we should also remove them on:

* actual remove
* disable

we have an issue already with the services though, but the services are Core only though?

Also I wonder if the undo of link-snap does the right thing? will it swap back the symlinks? or will just do nothing until we reach the undo of unlink-current-snap?